CDCgov/prime-reportstream
# 8659: clean auth code up now that organization store keys

This PR cleans up the api key now that organization stores public keys

Test Steps:

This query can be confirmed if you have a sender with keys associated or if you update a setting to set a key
```sql
with sender_id_with_keys as (
	select setting_id, "values"->'keys' as keys
	from setting
	where is_active = true and "type" = 'SENDER'
)
select s.name as setting_name, o."name"  as organizatinon_name, keys  from
setting s
join sender_id_with_keys using(setting_id)
join setting o on o.setting_id  = s.organization_id 
where keys != 'null'
```

- Verify that there no senders with keys in staging or prod
  1. Run the above query in stage and prod
  2. Verify that it does not return results
- Run the smoke test locally
- Add a key, generate a JWT and verify an access token can be generated
  - `./prime organization addkey --public-key "./examples/generate-jwt-python/my-rsa-public-key.pem" --scope "simple_report.*.report" --orgName "simple_report" --kid "simple_report.default4" --doit`
  - `./prime organization reqtoken --private-key ./examples/generate-jwt-python/my-rsa-keypair.pem --scope "simple_report.*.report" --name simple_report --kid simple_report.default4`



## Changes
- Removes references to senders having keys
- Updates OAuth server2server workflows to operate on orgs

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #8659

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
